python/lsst/cell_coadds/_identifiers.py

```diff
@@ -72,25 +94,39 @@ def from_data_id(cls, data_id: DataCoordinate) -> PatchIdentifiers:
         identifiers : `PatchIdentifiers`
             Struct of identifiers for this patch.
         """
-        # The 'type: ignore' directives below are present because we require a
-        # a fully-expanded data ID, but this isn't embedded in the types.
-        return cls(
-            skymap=data_id["skymap"],  # type: ignore
-            tract=data_id["tract"],  # type: ignore
-            patch=Index2D(
-                x=data_id.records["patch"].cell_x, y=data_id.records["patch"].cell_y  # type: ignore
-            ),
-            band=data_id.get("band"),
-        )
+        # This essentially calls the parent method with a specialized docstring
+        # and a restricted signature.
+        return super().from_data_id(data_id)
 
 
 @dataclass(frozen=True)
-class CellIdentifiers(PatchIdentifiers):
+class CellIdentifiers(BasePatchIdentifiers):
     """Struct of identifiers for a coadd cell."""
 
     cell: Index2D
     """Identifiers for the cell itself."""
 
+    @classmethod
+    def from_data_id(  # type: ignore [override]
+        cls, data_id: DataCoordinate, cell: Index2D
+    ) -> CellIdentifiers:
+        """Construct from a data ID and a cell index.
+
+        Parameters
+        ----------
+        data_id : `~lsst.daf.butler.DataCoordinate`
+            Fully-expanded data ID that includes the 'patch' dimension and
+            optionally the `band` dimension.
+        cell : `~lsst.skymap.Index2D`
+            Index of the cell within the patch.
+
+        Returns
+        -------
+        identifiers : `CellIdentifiers`
+            Struct of identifiers for this cell within a patch.
+        """
+        return super().from_data_id(data_id, cell=cell)
```

Comment:
I'm not sure the code reuse comes from inheriting this method is actually worth the `type: ignore` required by the `PatchIdentifiers` override.  I think it might be cleaner to drop the base class implementation and copy its implementation with appropriate modifications into the derived-class ones.